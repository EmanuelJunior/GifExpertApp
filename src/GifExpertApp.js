import React, { Fragment, useState } from "react"
import { AddCategory } from "./components/AddCategory"
import { GifGrid } from "./components/GifGrid"

const GifExpertApp = ({ defaultCategories = ['Rick and Morty'] }) => {

    const [categories, setCategories] = useState( defaultCategories );

    return (
        <Fragment>
            <h2>Gif Expert App</h2>

            <AddCategory setCategories={ setCategories }/>

            <hr></hr>

                {
                    categories.map( category =>  (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
        </Fragment>
    )
}

export default GifExpertApp;

