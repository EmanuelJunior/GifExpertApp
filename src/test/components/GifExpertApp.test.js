import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../../GifExpertApp';

describe('Pruebas en el componente <GifExpertApp />', ()=>{
    
    test('Debe de mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });  

    test('Debe de mostrar una lista de categorias', ()=> {
        const categories = ['One Punch','Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>)
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
})
