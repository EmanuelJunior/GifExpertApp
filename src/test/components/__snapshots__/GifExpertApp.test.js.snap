// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
